addons/calendar/models/calendar_event.py

```diff
@@ -324,6 +324,30 @@ def _compute_attendee(self):
             attendee = mapped_attendees[meeting.id]
             meeting.attendee_status = attendee.state if attendee else 'needsAction'
 
+    def _search_attendee_status(self, operator, value):
+        negative = operator in expression.NEGATIVE_TERM_OPERATORS
+
+        # In case we have no value or non supported operator
+        if not value or operator not in ['=', '!=', 'in', 'not in']:
+            return expression.TRUE_DOMAIN if negative else expression.FALSE_DOMAIN
+
+        all_status = set([status[0] for status in Attendee.STATE_SELECTION])
+        if operator == '=':
+            search_status = {value}
+        elif operator == '!=':
+            search_status = all_status - {value}
+        elif operator == 'in':
+            search_status = set(value)
+        elif operator == 'not in':
+            search_status = all_status - set(value)
+
+        self._cr.execute("""SELECT event_id FROM calendar_attendee WHERE state IN %s AND partner_id = %s""", (tuple(search_status), self.env.user.partner_id.id))
+        event_ids = [event[0] for event in self._cr.fetchall()]
+        self._cr.execute("""SELECT event_id FROM calendar_attendee WHERE id not in %s""", (tuple(event_ids), ))
```

Comment:
Why do you need this? Not sure I get it

Reply:
I did that to not filter out event that are not assigned to current user to achieve point ` "Non declined events" hides all the events from colleagues calendar which is not expected. It should only hide events for which the user is an attendee but declined. `
But this is completely wrong change to do that i'll update the code to achieve above point :slightly_smiling_face: